### PR TITLE
Add EnableSupportLogging feature

### LINKED ISSUE #34072: EnableSupportLogging (Allow extraction of all MSAL logs)
**Context:**
When customers encounter bugs in MSAL, the MSAL team needs detailed logs to diagnose the issue. The Azure Identity library doesn't have a provision to extract all these logs. In particular, we need verbose logs with PII information. For context, MSAL exposes 2 knobs for logging:

1. The log verbosity, which is very closely related to Azure SDK’s verbosity, i.e. Verbose, Info, Warning, Error.
2. A PII flag, false by default. Note: For a cheat sheet of what gets logged, see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/PII.

The request is to allow app developers using the Azure Identity library to collect Verbose + PII logs from MSAL, on request, which are sometimes needed for supportability. The Azure Identity library exposes some knobs already, but there's no alignment over PII.

### PR DESCRIPTION
**Pull request by billwert**
This enables users to turn on support logging for MSAL4j. Fixes #34072

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/AadCredentialBuilderBase.java [hunk 1]
```diff
@@ -126,4 +126,15 @@ public T disableInstanceDiscovery() {
         this.identityClientOptions.disableInstanceDiscovery();
         return (T) this;
     }
+
+    /**
+     * Enables additional support logging for public and confidential client applications. This enables
+     * PII logging in MSAL4J as described <a href="https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-logging-java#personal-and-organization-information">here.</a>
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/AadCredentialBuilderBase.java:132]: remove en-us from link

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java [hunk 2]
```diff
@@ -281,7 +284,9 @@ PublicClientApplication getPublicClient(boolean sharedTokenCacheCredential, bool
             + tenantId;
         PublicClientApplication.Builder builder = PublicClientApplication.builder(clientId);
         try {
-            builder = builder.authority(authorityUrl).instanceDiscovery(options.isInstanceDiscoveryEnabled());
+            builder = builder.
+                logPii(options.isSupportLoggingEnabled()).
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java:288]: nit: period goes on next line as well.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java [hunk 3]
```diff
@@ -395,7 +404,9 @@ ConfidentialClientApplication getWorkloadIdentityConfidentialClient() {
                 : clientId, credential);
 
         try {
-            applicationBuilder = applicationBuilder.authority(authorityUrl).instanceDiscovery(options.isInstanceDiscoveryEnabled());
+            applicationBuilder = applicationBuilder.authority(authorityUrl).
+                logPii(options.isSupportLoggingEnabled()).
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java:408]: nit; period goes on next line as well.

**Comment by billwert** [comment]
Remaining failure is azure-communication-rooms failing coverage. Unrelated to this PR.

**Review by g2vinay — APPROVED** [review]

**Pull request merged**